server/mscalendar/availability.go

```diff
@@ -276,19 +285,40 @@ func (m *mscalendar) setStatusFromCalendarView(user *store.User, currentStatus s
 	return message, nil
 }
 
-func (m *mscalendar) setStatusOrAskUser(user *store.User, status string, events []*remote.Event) error {
-	if user.Settings.GetConfirmation {
-		url := fmt.Sprintf("%s%s%s", m.Config.PluginURLPath, config.PathPostAction, config.PathConfirmStatusChange)
-		_, err := m.Poster.DMWithAttachments(user.MattermostUserID, views.RenderStatusChangeNotificationView(events, status, url))
-		if err != nil {
-			return err
+func (m *mscalendar) setStatusOrAskUser(user *store.User, status *model.Status, events []*remote.Event, isFree bool) error {
+	toSet := model.STATUS_ONLINE
+	if isFree && user.LastStatus != "" {
+		toSet = user.LastStatus
+		user.LastStatus = ""
+	}
+
+	if !isFree {
+		toSet = model.STATUS_DND
+		if user.Settings.ReceiveNotificationsDuringMeeting {
+			toSet = model.STATUS_AWAY
+		}
+		if !user.Settings.GetConfirmation && status.Manual {
+			user.LastStatus = status.Status
+		}
+	}
+
+	err := m.Store.StoreUser(user)
+	if err != nil {
+		return err
+	}
+
+	if !user.Settings.GetConfirmation {
+		_, appErr := m.PluginAPI.UpdateMattermostUserStatus(user.MattermostUserID, toSet)
```

Comment:
So if the user has asked to get confirmation changes, they also get asked after the meeting? That makes sense, as then we are not ever changing the user's status without their approval.

@asaadmahmood @aaronrothschild Thoughts on this?

Reply:
I thought this was like this by design from the beginning. As far as I know, it has always been implemented this way.